src/main/java/ghostcat/loginLambda/Login.java

```diff
@@ -0,0 +1,31 @@
+package ghostcat.loginLambda;
+
+import java.security.SecureRandom;
+import java.time.Instant;
+
+public class Login {
+  public LoginResult handleRequest(LoginRequest request) {
+    LoginResult result = new LoginResult();
+    LoginDAO dao = new LoginDAO();
+
+    if (dao.valid(request.getUserID(), request.getPasswordHash())) {
+      String AB = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+      SecureRandom rnd = new SecureRandom();
+      StringBuilder stringBuilder = new StringBuilder();
+      for (int i = 0; i < 10; ++i) {
+        stringBuilder.append( AB.charAt( rnd.nextInt(AB.length()) ) );
+      }
+      StringBuilder time = new StringBuilder();
+      time.append(Instant.now().toEpochMilli());
+      stringBuilder.append(time.toString());
+      String auth = stringBuilder.toString();
+      dao.addAuth(request.userID, auth, time.toString());
+      result.setAuth_token(auth);
```

Comment:
Is the auth token here just the password hash with the salt appended? I don't think that exposing the password hash as the auth token is the best idea. The other problem is that the auth token would be the same for a given user every time they logged in. Instead it might be best to do something like this to create the auth token:
`byte[] randomBytes = new byte[24];`
`secureRandom.nextBytes(randomBytes);`
`String token = base64Encoder.encodeToString(randomBytes);`

Reply:
Agreed. Can we use UUID for the authtoken?

Reply:
I'll get on that